dbos/_dbos.py

```diff
@@ -320,6 +318,7 @@ def __init__(
 
         self._initialized: bool = True
 
+        self._provided_config: DBOSConfig = config
```

Comment:
Get rid of this variable and the check above where it's used. Confusing and not entirely consistent (the check only checks some of the arguments, not all of them).